### ISSUE TITLE
compiler crash with new-ES6 object used in enum

### ISSUE THREAD

**Issue by nicks** [google/closure-compiler#1448]
Code:
```js
/** @enum */ var X = {F};
```

crash!
```
java.lang.RuntimeException: INTERNAL COMPILER ERROR.
Please email js-compiler@google.com with this stack trace and repro steps (a CL# and a build target).

Unexpected variable F
  Node(NAME F): input0:2:22
/** @enum */ var X = {F};
  Parent(STRING_KEY F): input0:2:22
/** @enum */ var X = {F};

	at com.google.javascript.jscomp.VarCheck.visit(VarCheck.java:204)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:625)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:619)
	at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:619)
```

http://closure-compiler-debugger.appspot.com/#input0%3D%250A%252F**%2520%2540enum%2520*%252F%2520var%2520X%2520%253D%2520%257BF%257D%253B%26input1%26conformanceConfig%26externs%26refasterjs-template%26includeDefaultExterns%3D1%26CHECK_SYMBOLS%3D1%26CHECK_TYPES%3D1%26CLOSURE_PASS%3D1%26LANG_IN_IS_ES6%3D1%26MISSING_PROPERTIES%3D1%26PRESERVE_TYPE_ANNOTATIONS%3D1%26PRETTY_PRINT%3D1%26TRANSPILE%3D1

**Comment by concavelenz** [comment]
for some reason an implicit extern is not being created for the undeclared var "F" reference here.  Really, CheckSymbols should be reporting an error.
